bin/live/pycbc_live_combine_single_fits

```diff
@@ -58,30 +58,33 @@ for f in files:
     if 'days_since_epoch' in fits_f.attrs:
         analysis_dates += [fits_f.attrs['days_since_epoch']]
     elif 'analysis_date' in fits_f.attrs:
-        analysis_year = int(fits_f.attrs['analysis_date'][:4])
-        analysis_month = int(fits_f.attrs['analysis_date'][5:7])
-        analysis_day = int(fits_f.attrs['analysis_date'][8:10])
+        date_array = fits_f.attrs['analysis_date'].split('_')
+        analysis_year = int(date_array[0])
+        analysis_month = int(date_array[1])
+        analysis_day = int(date_array[2])
         n_since_epoch = (datetime.date(analysis_year,
                                        analysis_month,
                                        analysis_day)
                          - datetime.date(2000, 1, 1)).days
         analysis_dates += [n_since_epoch]
     else:
-        # This is the regex string to match a date in format YYYY_MM_DD
+        # This is the regex string to match a date in format YYYY_MM_DD, 
+        # or if MM or DD only have one character
         # FutureWarning: this will become obsolete in the year 3000
-        re_string = '([12]\d{3}_(0[1-9]|1[0-2])_(0[1-9]|[12]\d|3[01]))'
+        re_string = '([12]\d{3}_(0?[1-9]|1[0-2])_(0?[1-9]|[12]\d|3[01]))'
         m = re.search(re_string, f)
-        analysis_year = int(m.group(0)[:4])
-        analysis_month = int(m.group(0)[5:7])
-        analysis_day = int(m.group(0)[8:10])
+        date_array = m.group(0).split('_')
+        analysis_year = int(date_array[0])
+        analysis_month = int(date_array[1])
+        analysis_day = int(date_array[2])
```

Comment:
I am a bit confused by the parentheses in the RE and by the split on underscores. Can you not just do this:
```python
re_string = '([12]\d{3})_(0?[1-9]|1[0-2])_(0?[1-9]|[12]\d|3[01])'
m = re.search(re_string, f)
analysis_year = int(m.group(1))
analysis_month = int(m.group(2))
analysis_day = int(m.group(3))
```

Reply:
that is something I didn't know you could do. That makes more sense

Reply:
Makes more sense now, but if my parsing of the RE is correct, should it not be
```
([12]\d{3})_(0?[1-9]|1[0-2])_(0?[1-9]|[12]\d|3[01])
```
instead of
```
([12]\d{3}_(0?[1-9]|1[0-2])_(0?[1-9]|[12]\d|3[01]))
```
(a subtle difference in parentheses)?